### DISCUSSION TITLE
During the process of using DeepMD kit, how can we improve the GPU utilization?

### DISCUSSION THREAD

**Discussion by PolyuWeldingSpock** [deepmodeling/deepmd-kit#2948] [General]
During the process of training my data with DeepMD kit, I noticed that the GPU utilization is only 60%. May I ask if there are any parameter settings that can enhance the GPU utilization during the DeepMD kit training process? My configuration is 6 CPUs + 64GB memory, paired with a 3080Ti graphics card.
![5bf8e3f2e896bb41a3afb1ac049b4b20](https://github.com/deepmodeling/deepmd-kit/assets/122458488/527deec4-79c0-4a33-bf26-fe6ec3eec84f)

**Comment by njzjz** [comment]
Increasing the batch size may help. Have you used the latest version?

**Reply by PolyuWeldingSpock** [reply]
Hi @njzjz ,  I have set batch_size to"auto".  After updating deepmd kit to v2.2.7, the utilization have increased to 70%.

**Reply by njzjz** [reply]
We've not released v2.2.7. Do you mean the development version?

**Reply by PolyuWeldingSpock** [reply]
@njzjz sorry about my answer. Yes, I mean the dev version

**Reply by njzjz** [reply]
If you built from source, I want to confirm if you have enabled GPU support by setting `export DP_VARIANT=cuda` during installation.

**Reply by PolyuWeldingSpock** [reply]
I installed with `conda`  not from source.

**Reply by njzjz** [reply]
The GPU utilization is the percentage of the GPU time in the total time. You can enable [`enable_profiler`](https://docs.deepmodeling.com/projects/deepmd/en/master/train/train-input.html#argument:training/enable_profiler) for profiling.

**Reply by PolyuWeldingSpock** [reply]
OK, Thanks for your help.
